packages/contracts/hardhat.config.ts

```diff
@@ -49,11 +49,22 @@ import "solidity-coverage";
 // read MNEMONIC from file or from env variable
 let mnemonic = process.env.MNEMONIC;
 
-const networkConfig = (chain: Chain): HttpNetworkUserConfig => ({
-    chainId: chain.id,
-    url: chain.rpcUrls.public.http.at(0),
-    accounts: mnemonic ? { mnemonic } : undefined,
-});
+const networkConfig = (chain: Chain): HttpNetworkUserConfig => {
+    let url = chain.rpcUrls.public.http.at(0);
+
+    // support for infura and alchemy URLs through env variables
+    if (process.env.INFURA_ID && chain.rpcUrls.infura?.http) {
+        url = `${chain.rpcUrls.infura.http}/${process.env.INFURA_ID}`;
+    } else if (process.env.ALCHEMY_ID && chain.rpcUrls.alchemy?.http) {
+        url = `${chain.rpcUrls.alchemy.http}/${process.env.INFURA_ID}`;
```

Comment:
Why `INFURA_ID` and not `ALCHEMY_ID`?

Reply:
Because the public URL not always work. So I added an option to use infura or alchemy.
Not all networks are available on infura or alchemy though.

Reply:
Ok, but you're using `INFURA_ID` with Alchemy's URL. Is that intended?

Reply:
No, it’s a mistake 🙏